android/src/main/java/com/reactnativespokestack/SpokestackModule.kt

```diff
@@ -19,6 +20,14 @@ class SpokestackModule(private val reactContext: ReactApplicationContext): React
   private var say: ((url: String) -> Unit)? = null
   private lateinit var audioPlayer: SpokestackTTSOutput
   private lateinit var downloader: Downloader
+  private val filenameToProp = mutableMapOf(
```

Comment:
Kotlin has a [`Pair`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-pair/) that would make this downloading a bit less painful:

```
private val downloadMap = mapOf(
  "filter" to Pair("filter.tflite", "wake-filter-path"),
  // ...
)
```

...I just spent far too long trying to rig up a `fold` or `reduce` snippet to make the actual transformation look prettier, but I give up on functional Kotlin for now.

Reply:
Nice 👍

Reply:
I looked into this, but it became a little awkward and then about as much code when I realized that downloadMap["filter"] had to be treated as optional. I could have simplified it further if I was willing to share this map with the Downloader, but that would break separation of concerns.

Reply:
I was mainly trying to get rid of the filters down in the key conversion code:
`wakeDownloads[filenameToProp.keys.first { it.contains("filter") }]`
Having that filter inside a map key just hurt me a bit.

Reply:
What I really want is a 2-way map.

Reply:
If you want that, you could always just make 2 maps:
```
private val propToFilename = filenameToProp.entries.associateBy({it.value}, {it.key})
```